sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/LiteralGenerator.scala

```diff
@@ -123,6 +124,18 @@ object LiteralGenerator {
       yield Literal.create(new Date(day * MILLIS_PER_DAY), DateType)
   }
 
+  private def timeMicrosGen: Gen[Long] = {
+    // Generate a random time represented as microseconds since midnight.
+    val minMicros = localTime()  // 00:00:00.000000
+    val maxMicros = localTime(23, 59, 59, 999999)  // 23:59:59.999999 in microseconds
+    Gen.choose(minMicros, maxMicros)
+  }
+
+  lazy val timeTypeLiteralGen: Gen[Literal] = {
+    for { micros <- timeMicrosGen }
+      yield Literal.create(micros, TimeType(6))  // Use microsecond precision
+  }
```

Comment:
There is a separate task for this: https://issues.apache.org/jira/browse/SPARK-51455. How about to implement it in a separate PR, and test the existing expression: `ToTime`?

Reply:
Ahh didn't notice the separate JIRA. I'll cater this change for that jira then.